images/build-drivers/build-drivers.sh

```diff
@@ -58,37 +54,27 @@ function start_docker() {
 	echo "Done setting up docker in docker."
 }
 
-function build() {
-	PUBLISH_S3="${PUBLISH_S3:-false}"
-	export PULL_PULL_SHA=$PULL_PULL_SHA
-	
+function build_and_publish() {
 	for filter_key in "${!DBG_FILTERS[@]}"; do
 		test -z "${DBG_FILTERS[$filter_key]}" \
-			|| DBG_MAKE_BUILD_OPTIONS="${DBG_MAKE_BUILD_OPTIONS} -e ${filter_key}=${DBG_FILTERS[$filter_key]}"
+			|| DBG_MAKE_BUILD_OPTIONS="${DBG_MAKE_BUILD_OPTIONS} ${filter_key}=${DBG_FILTERS[$filter_key]}"
 	done
-
-	pretty_echo "Testing building Drivers for PR hash $PULL_PULL_SHA"
 	
-	cd $DBG_WORKDIR
-	touch output/failing.log
-	DBG_COMMIT=$(git log -1 --format=format:%H --full-diff -- ./)
-
-	pretty_echo "Found current driverkit version DBG_COMMIT $DBG_COMMIT. Running DBG build..."
-	$make $DBG_MAKE_BUILD_OPTIONS $DBG_MAKE_BUILD_TARGET
+	pretty_echo "Running DBG build..."
+	if [[ "$DBG_MAKE_BUILD_TARGET" -eq "build" ]]; then
+		# when building, ignore errors (just report them back to driverkit/output/failing.log)
+		DBG_MAKE_BUILD_OPTIONS="${DBG_MAKE_BUILD_OPTIONS} --ignore-errors --skip-existing --redirect-errors=driverkit/output/failing.log"
+		# If requested, publish too!
+		test "${PUBLISH_S3}" == "true" && DBG_MAKE_BUILD_OPTIONS="${DBG_MAKE_BUILD_OPTIONS} --publish"
```

Comment:
`dbg-go` supports publishing while building (ie: asynchronously in a goroutine while the next build is being done).
This has 2 advantages:
* drivers will become available right after being built (without wait all the drivers to be built before start publishing)
* time spent uploading drivers is saved, since it is done async

Reply:
> time spent uploading drivers is saved, since it is done async

Even if it does not seem that big of a deal, imagine if we take like 1s to upload a single driver; and imagine we upload like 2000 drivers each week (i haven't got any data about it atm); we are saving 2000 seconds, ie ~30mins!

Reply:
This is great